hoomd/md/pair/pair.py

```diff
@@ -2141,3 +2141,58 @@ def __init__(self, nlist, r_cut=None, r_on=0., mode='none'):
                                    k=float, phi=float, len_keys=2)
                                )
         self._add_typeparam(params)
+
+class TWF(Pair):
+    r"""Pair potential model for globular proteins.
+
+    This potential was introduced by Ten-wolde and Daan Frenkel in 1997 for
+    studying globular protein crystallization. The potential has the following
+    form:
+
+    .. math::
+        :nowrap:
+
+        \\begin{equation}
+        V_{\\mathrm{TWF}}(r) = \\frac{4 \\epsilon}{\\alpha^2} {\\left[
+        {\\left(\\frac{\\sigma^2}{r^2} - 1 \\right)}^6 -
+        \\alpha {\\left(\\frac{\\sigma^2}{r^2} - 1 \\right)}^3\\right]}
+        \\end{equation}
+
+    See `hoomd.md.pair.Pair` for details on how forces are calculated and the
+    available energy shifting and smoothing modes.  Use `params` dictionary
+    to set potential coefficients. The coefficients must be set per
+    unique pair of particle types.
+
+    The potential comes from
+    `Pieter Rein ten Wolde and Daan Frenkel 1997 <https://dx.doi.org/10.1126/science.277.5334.1975 >`_.
+
+    Attributes:
+        params (`TypeParameter` [\
+            `tuple` [``particle_type``, ``particle_type``],\
+            `dict`]):
+            The LJ potential parameters. The dictionary has the following keys:
+
+            * ``epsilon`` (`float`, **required**) -
+              energy parameter :math:`\\varepsilon` (in energy units)
+
+            * ``sigma`` (`float`, **required**) -
+              particle size :math:`\\sigma` (in distance units)
```

Comment:
```suggestion
              particle size :math:`\\sigma` (in length units)
```
We need to get better about using length and distance properly throughout the code.

Reply:
Agreed